includes/class-google-extendedaccess.php

```diff
@@ -68,9 +72,62 @@ public static function add_extended_access_ld_json() {
 				'publisher'           => array(
 					'@type' => 'Organization',
 					'name'  => get_bloginfo( 'name' ),
+					'url'   => home_url(),
+					'logo'  => [
+						'@type'  => 'ImageObject',
+						'url'    => get_site_icon_url(),
+					],
 				),
 			);
 
+			// Dates in ISO 8601 format as per schema standards.
+			$ld_json['datePublished'] = get_the_date( 'c' );
+			$ld_json['dateModified']  = get_the_modified_date( 'c' );
+
+			$post_authors      = [];
+			$ld_json['author'] = [ // Default to the organization.
+				'@type' => 'Organization',
+				'name'  => get_bloginfo( 'name' ),
+				'url'   => home_url(),
+			];
+			if ( function_exists( 'get_coauthors' ) ) {
+				$post_authors = get_coauthors();
+			}
+			if ( empty( $post_authors ) ) {
+				$author_id    = absint( get_the_author_meta( 'ID' ) );
+				$post_authors = array( get_userdata( $author_id ) );
+			}
+			if ( ! empty( $post_authors ) ) {
+				$authors = array();
+				foreach ( $post_authors as $author ) {
+					if ( $author ) {
+						$author_data = [
+							'@type' => 'Person',
+							'name'  => $author->display_name,
+						];
+
+						$author_url = get_author_posts_url( $author->ID );
+						if ( $author_url ) {
+							$author_data['url'] = $author_url;
+						}
+
+						$authors[] = $author_data;
+					}
+				}
+
+				if ( ! empty( $authors ) ) {
+					$ld_json['author'] = count( $authors ) === 1 ? $authors[0] : $authors;
+				}
+			}
+
+			if ( has_post_thumbnail() ) {
+				$image_url = get_the_post_thumbnail_url();
+				if ( $image_url ) {
+					$ld_json['thumbnailUrl'] = $image_url;
+					$ld_json['image']        = $image_url; // This could be any image or array of images, not just the thumbnail.
```

Comment:
```suggestion
				$image_sizes  = [
					[ '400', '225' ], // 16:9.
					[ '320', '240' ], // 4:3.
					[ '250', '250' ], // 1:1.
				];
				foreach ( $image_sizes as $size ) {
					$ld_json['image'][] = wp_get_attachment_image_url( get_post_thumbnail_id(), $size );
				}
```

We can also pass the images google suggests like this.
See: [Google's docs](https://developers.google.com/search/docs/appearance/structured-data/article#:~:text=For%20best%20results%2C%20we%20recommend%20providing%20multiple%20high%2Dresolution%20images%20(minimum%20of%2050K%20pixels%20when%20multiplying%20width%20and%20height)%20with%20the%20following%20aspect%20ratios%3A%2016x9%2C%204x3%2C%20and%201x1.)